File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        n = len(nums)//2
        ans = 1
        d = {}
        
        for i in nums:
            if i not in d.keys():
                d[i] = 1
            else:
                d[i] += 1
                if d[i]>n: 
                    ans = i
        return ans
            
        

File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int n = nums.size();
        int candidate = nums[0];
        int votes = 1;
        
        for(int i=0; i<n; i++){
            if(nums[i]==candidate) votes++;
            else{
                if(nums[i]!=candidate) votes--;
                if(votes==0){
                    candidate = nums[i];
                    votes = 1;
                } 
            }
        }

        
        return candidate;
        
    }
};